apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+    this.sentAudioKey = 'message.sent-sound.enabled';
+    this.sentAudioEnabled = false;
+
+    // navigator.mozSettings may not be defined in all environments
+    if (navigator.mozSettings) {
+      try {
+        var req = navigator.mozSettings.createLock().get(this.sentAudioKey);
+        req.onsuccess = (function onsuccess() {
+          this.sentAudioEnabled = req.result[this.sentAudioKey];
+        }).bind(this);
+
+        navigator.mozSettings.addObserver(this.sentAudioKey, (function(e) {
+          this.sentAudioEnabled = e.settingValue;
+        }).bind(this));
+      } catch (e) {
+        this.sentAudioEnabled = false;
+      }
+    }
+  },
+
+  getIdIterator: function conv_getIdIterator() {
+    return Threads.active.messages.keys();
+  },
+
+  setHeaderAction: function conv_setHeaderAction(icon) {
+    this.header.setAttribute('action', icon);
+  },
+
+  messageComposerInputHandler: function conv_messageInputHandler(event) {
+    if (Compose.type === 'sms') {
+      this.hideMaxLengthNotice();
+      return;
+    }
+
+    if (Compose.isResizing) {
+      this.resizeNotice.classList.remove('hide');
+
+      if (this._resizeNoticeTimeout) {
+        clearTimeout(this._resizeNoticeTimeout);
+        this._resizeNoticeTimeout = null;
+      }
+    } else {
+      this.checkMessageSize();
+      if (this.resizeNotice.classList.contains('hide') ||
+          this._resizeNoticeTimeout) {
+        return;
+      }
+
+      this._resizeNoticeTimeout = setTimeout(function hideResizeNotice() {
+        this.resizeNotice.classList.add('hide');
+        this._resizeNoticeTimeout = null;
+      }.bind(this), this.IMAGE_RESIZE_DURATION);
+    }
+  },
+
+  showMaxLengthNotice: function conv_showMaxLengthNotice(opts) {
+    Compose.lock();
+    navigator.mozL10n.setAttributes(
+      this.maxLengthNotice.querySelector('p'), opts.l10nId, opts.l10nArgs
+    );
+    this.maxLengthNotice.classList.remove('hide');
+  },
+
+  hideMaxLengthNotice: function conv_hideMaxLengthNotice() {
+    Compose.unlock();
+    this.maxLengthNotice.classList.add('hide');
+  },
+
+  showSubjectMaxLengthNotice: function conv_showSubjectMaxLengthNotice() {
+    this.subjectMaxLengthNotice.classList.remove('hide');
+
+    if (this.timeouts.subjectLengthNotice) {
+      clearTimeout(this.timeouts.subjectLengthNotice);
+    }
+    this.timeouts.subjectLengthNotice = setTimeout(
+      this.hideSubjectMaxLengthNotice.bind(this),
+      this.BANNER_DURATION
+    );
+  },
+
+  hideSubjectMaxLengthNotice: function conv_hideSubjectMaxLengthNotice() {
+    this.subjectMaxLengthNotice.classList.add('hide');
+    this.timeouts.subjectLengthNotice &&
+      clearTimeout(this.timeouts.subjectLengthNotice);
+  },
+
+  /*
+   * This function will be called before we slide to the thread or composer
+   * panel. This way it can change things in the panel before the panel is
+   * visible.
+   */
+  beforeEnter: function conv_beforeEnter(args) {
```

Comment:
> But we may need it at first since we'll use the same JS both in split and non-split modes.

Yeah, I'm currently thinking how to handle this, what I meant is that we don't have think too much about this piece since it will be removed :)

> So how about navigating to another conversation thread?

You mean conversation#1---->conversation#2 case when we separate views? If yes that's a good question, we talked with @julienw about this some time ago - whether we want to have real navigation from say ```..../conversation/?id=1``` to ```..../conversation/?id=2```  or just update current view with new conversation, at that time I was thinking about the latter, but now I think real navigation is more preferable :)